docs/fundamentals/networking/telemetry/event-counters.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Networking Event Counters in .NET
+description: Learn how to consume .NET networking event counters.
+author: MihaZupan
+ms.author: mizupan
+ms.date: 10/18/2022
+---
+
+# Networking Event Counters in .NET
+
+[EventCounters] are .NET APIs used for lightweight, cross-platform, and near real-time performance metric collection.
+
+Networking components are instrumented to publish basic diagnostics information using EventCounters.
+They include information like the following:
+
+- `System.Net.Http` > `requests-started`
+- `System.Net.Http` > `requests-failed`
+- `System.Net.Http` > `http11-connections-current-total`
+- `System.Net.Security` > `all-tls-sessions-open`
+- `System.Net.Sockets` > `outgoing-connections-established`
+- `System.Net.NameResolution` > `dns-lookups-duration`
+
+> [!TIP]
+> For the full list, see [well-known counters].
+
+> [!TIP]
+> On projects targeting .NET 8+, consider using the newer and more feature-rich [networking metrics] instead of EventCounters.
+
+## Providers
+
+Networking information is split across the following providers:
+
+- `System.Net.Http` (`HttpClient` and `SocketsHttpHandler`)
+- `System.Net.NameResolution` (`Dns`)
+- `System.Net.Security` (`SslStream`)
+- `System.Net.Sockets`
+- `Microsoft.AspNetCore.Hosting`
+- `Microsoft-AspNetCore-Server-Kestrel`
+
+The telemetry has some performance overhead when enabled, so make sure to subscribe only to providers you're actually interested in.
+
+## Monitor Event Counters from outside the process
+
+### dotnet-counters
+
+[`dotnet-counters`][dotnet-counter docs] is a cross-platform performance monitoring tool for ad-hoc health monitoring and first-level performance investigation.
+
+```console
+dotnet tool install --global dotnet-counters
+```
+
+```console
+dotnet-counters monitor --counters System.Net.Http,System.Net.Security --process-id 1234
+```
+
+The command continually refreshes the console with the latest numbers.
+
+```txt
+[System.Net.Http]
+    Current Http 1.1 Connections                       3
+    Current Http 2.0 Connections                       1
+    Current Http 3.0 Connections                       0
+    Current Requests                                   4
+    HTTP 1.1 Requests Queue Duration (ms)              0
+    HTTP 2.0 Requests Queue Duration (ms)              0
+    HTTP 3.0 Requests Queue Duration (ms)              0
+    Requests Failed                                    0
+    Requests Failed Rate (Count / 1 sec)               0
+    Requests Started                                 470
+    Requests Started Rate (Count / 1 sec)             18
+```
+
+For all the available commands and parameters, see the [dotnet-counter docs].
+
+### Application Insights
+
+Application Insights does not collect event counters by default.
+For information on customizing the set of counters you're interested in, see the [AppInsights EventCounters docs].
+
+For example:
+
+```c#
```

Comment:
Can you change all the instances of lowercase `c#` to `C#`? Otherwise it renders as "c#". See https://review.learn.microsoft.com/en-us/dotnet/fundamentals/networking/telemetry/events?branch=pr-en-us-38422#eventlistener.

Reply:
Done.